src/envoy/http/backend_auth/config_parser_impl.cc

```diff
@@ -55,26 +56,31 @@ AudienceContext::AudienceContext(
           filter_config.iam_token().iam_uri().cluster();
       const std::chrono::seconds fetch_timeout(TimeUtil::DurationToSeconds(
           filter_config.iam_token().iam_uri().timeout()));
+      const DependencyErrorBehavior error_behavior =
+          filter_config.dep_error_behavior();
       const std::string real_uri =
           absl::StrCat(uri, "?audience=", jwt_audience);
       const ::google::protobuf::RepeatedPtrField<std::string>& delegates =
           filter_config.iam_token().delegates();
       iam_token_sub_ptr_ = token_subscriber_factory.createIamTokenSubscriber(
-          TokenType::IdentityToken, cluster, real_uri, fetch_timeout, callback,
-          delegates, ::google::protobuf::RepeatedPtrField<std::string>(),
-          access_token_fn);
+          TokenType::IdentityToken, cluster, real_uri, fetch_timeout,
+          error_behavior, callback, delegates,
+          ::google::protobuf::RepeatedPtrField<std::string>(), access_token_fn);
     }
       return;
     case FilterConfig::IdTokenInfoCase::kImdsToken: {
       const std::string& uri = filter_config.imds_token().uri();
       const std::string& cluster = filter_config.imds_token().cluster();
       const std::chrono::seconds fetch_timeout(
           TimeUtil::DurationToSeconds(filter_config.imds_token().timeout()));
+      const DependencyErrorBehavior error_behavior =
+          filter_config.dep_error_behavior();
       const std::string real_uri =
           absl::StrCat(uri, "?format=standard&audience=", jwt_audience);
 
       imds_token_sub_ptr_ = token_subscriber_factory.createImdsTokenSubscriber(
-          TokenType::IdentityToken, cluster, real_uri, fetch_timeout, callback);
+          TokenType::IdentityToken, cluster, real_uri, fetch_timeout,
```

Comment:
The code LGTM. 

Not sure if it is already in your plan, one thing related to this use case for is api producer doesn't know the rejection is caused by initialization failure except access log, which is hidden for API Gateway. It is necessary the info of dependency failure(or more specifically IAM/IMDS failure) into access log / report(maybe not filter stats) for better correlating.

Reply:
Ack. I agree, currently the API Producer will not easily know what the error is unless they look at the application log. I need to think of how this can be reported better.